crates/core/src/bc_protocol/connection/udpconn/discover.rs

```diff
@@ -0,0 +1,543 @@
+                    mtu,
+                    debug: false,
+                    os: "MAC".to_string(),
+                }),
+                ..Default::default()
+            },
+        });
+
+        let mut buf = vec![];
+        msg.serialize(&mut buf)?;
+
+        let mut broadcasts = vec![Ipv4Addr::BROADCAST];
+        for iface in get_if_addrs::get_if_addrs()?.iter() {
+            if let get_if_addrs::IfAddr::V4(ifacev4) = &iface.addr {
+                if let Some(broadcast) = ifacev4.broadcast.as_ref() {
+                    broadcasts.push(*broadcast);
+                }
+            }
+        }
+        let ports: [u16; 2] = [2015, 2018];
+        let destinations: Vec<(Ipv4Addr, u16)> = broadcasts
+            .iter()
+            .map(|&addr| {
+                ports
+                    .iter()
+                    .map(|&port| (addr, port))
+                    .collect::<Vec<(Ipv4Addr, u16)>>()
+            })
+            .flatten()
+            .collect();
+        debug!("Broadcasting to: {:?}", destinations);
+
+        let (amount_sent, abort) = Self::retrying_send_to_multi(socket, &buf[..], &destinations)?;
+        assert_eq!(amount_sent, buf.len());
+
+        let start_time = OffsetDateTime::now_utc();
+        let camera_address;
+        let camera_id;
+        loop {
+            if (OffsetDateTime::now_utc() - start_time) <= timeout {
+                let mut buf = vec![0; mtu as usize];
+                if let Ok((bytes_read, socket_addr)) = socket.recv_from(&mut buf) {
+                    match BcUdp::deserialize(&buf[0..bytes_read]) {
+                        Ok(BcUdp::Discovery(UdpDiscovery {
+                            tid: _,
+                            payload:
+                                UdpXml {
+                                    d2c_c_r: Some(D2cCr { did, cid, .. }),
+                                    ..
+                                },
+                        })) if cid == client_id => {
+                            abort.abort();
+                            camera_address = socket_addr;
+                            camera_id = did;
+                            break;
+                        }
+                        Ok(smt) => debug!("Udp Discovery got this unexpected BcUdp {:?}", smt),
+                        Err(_) => debug!(
+                            "Udp Discovery got this unexpected binary: {:?}",
+                            &buf[0..bytes_read]
+                        ),
+                    }
+                }
+            } else {
+                abort.abort();
+                return Err(Error::Timeout);
+            }
+        }
+
+        // Ensure we didn't forget to abort the sending
+        abort.abort();
+        // Actual connection time!
+        socket.connect(camera_address)?;
+
+        Ok(UdpDiscover {
+            address: camera_address,
+            client_id,
+            camera_id,
+            mtu,
+        })
+    }
+
+    fn get_register(
```

Comment:
```suggestion
    // This function will contact the p2p relay servers
    //
    // It will ask each of the servers for details on a specific UID
    //
    // On success it returns the M2cQr that the p2p relay
    // server has about the UID
    fn get_register(
```